### DISCUSSION TITLE
An issue with dense offset step with alos2App.py

### DISCUSSION THREAD

**Discussion by y-okur** [isce-framework/isce2#674] [Q&A] [answered]
I was able to do dense offset analysis with ALOS2 stripmap data (FBD mode) but when I tried to do it with ScanSAR (WBS mode), dense offset step was skipped. I was wondering how I can do this with ScanSAR data?

**Comment by EJFielding** [comment]
I don't think it is feasible to calculate accurate dense offsets on the ALOS-2 ScanSAR (wide-swath) SLC (level 1.1) data focussed with the full-aperture method. I believe the "alos2App.py" code is set to skip dense offsets even if you request it for ScanSAR data. See these papers for the difficulties with cross-correlations on the full-aperture SLC images:
```
Liang, C., and E. J. Fielding (2016). Interferometric Processing of ScanSAR Data Using Stripmap Processor: New Insights From Coregistration, IEEE Transactions on Geoscience and Remote Sensing 54, no. 7, 4343-4354, doi:10.1109/TGRS.2016.2539962.

Liang, C., and E. J. Fielding (2017). Interferometry With ALOS-2 Full-Aperture ScanSAR Data, IEEE Transactions on Geoscience and Remote Sensing 55, no. 5, 2739-2750, doi:10.1109/TGRS.2017.2653190.
```

**Comment by CunrenLiang** [comment, marked as answer]
In general there is no big need for ALOS-2 ScanSAR offsets. Range displacement can be calculated using regular InSAR, while azimuth displacement can be calculated using alos2burstApp.py. Both of them are more precise than offsets in the respective directions. So we did not add the capability in alos2App.py to estimate ScanSAR offsets.

But if you prefer offsets, here is how you can do it.

(a) if you only need range offset, you only need to change the following scripts
isce2-main/components/isceobj/Alos2Proc/runSlcMatch.py
isce2-main/components/isceobj/Alos2Proc/runSlcMosaic.py
isce2-main/components/isceobj/Alos2Proc/runDenseOffset.py
isce2-main/components/isceobj/Alos2Proc/runFiltOffset.py
isce2-main/components/isceobj/Alos2Proc/runGeocodeOffset.py

comment out the following two lines in those scripts.
```
    if not ((self._insar.modeCombination == 0) or (self._insar.modeCombination == 1)):
        return
```

(b) if you also need azimuth offset, you also need to change
isce2-main/components/mroipac/ampcor/DenseAmpcor.py
at line 471, add line 215-216 from the following script
isce2-main/components/isceobj/Alos2Proc/runSlcOffset.py
and change variable name from ampcor to objAmpcor

setting the parameter at line 370 in input paramter file
examples/input_files/alos2/alos2App.xml
using value at line 210 in
isce2-main/components/isceobj/Alos2Proc/runSlcOffset.py

setting the parameter at line 380 in input paramter file
```
<property name="dense offset search window hight">64</property>
```

setting the parameter at line 146 in input paramter file
```
    <property name="use GPU">False</property>
```

Currently we can only compute offsets swath by swath. You can geocode the results and mosaic them using other software e.g. GMT. There might be constant offset between swaths.
So the following paramters in input parameter file should be the same.
```
    <property name="starting swath">None</property>
    <property name="ending swath">None</property>
```
frames may be mosaicked.

**Reply by y-okur** [reply]
Dear CunrenLiang,

Thank you very much for your reply. I tried alos2burstApp.py and now I have the azimuth displacements. I didn't understand what you meant by _"Range displacement can be calculated using regular InSAR"_ to obtain range displacement.

**Reply by fdelgadodelapuente** [reply]
Hi y-okur,

Range displacement can be measured by both standard InSAR and range offsets. The difference is that offsets are noisier than phase for slant range deformation, so you need data with a small pixel size (like CSK or TSX) or very large strain that will saturate the phase and will render unwrapping impossible.

**Reply by y-okur** [reply]
Hi @CunrenLiang

I was able to obtain the range offsets by using the method that you proposed. I looked at the track.xml file to find _"rangepixelsize"_ to convert range offsets to meters, but _"rangepixelsize"_  wasn't written on track.xml. Is there a way to calculate it?

**Reply by y-okur** [reply]
I found the _"rangepixelsize"_ in the yymmdd.frame.xml files in the f1_**** folders.

**Reply by EJFielding** [reply]
Yes, I was going to write that you should look at the "$date.frame.xml" files. The JAXA ALOS-2 ScanSAR (WD1) images are mostly acquired at 14 MHz range bandwidth, so the range pixel spacing is quite large, around 8.6 meters. Because of the full-aperture focusing method of the ALOS-2 ScanSAR products, the azimuth pixel spacing is small (around 2.6 meters) but the actual azimuth resolution is much larger (around 40 meters). I don't think pixel offsets in the azimuth direction will ever be useful from the ScanSAR full-aperture SLC images.

**Reply by y-okur** [reply]
@EJFielding, thank you very much for the reply. I analyzed the same region using ALOS-2 ScanSAR and Stripmap using dense offsets. Range offsets from both products showed similar results, but ScanSAR was noisier. However, the azimuth direction from ScanSAR didn't seem useful at all, just like you said.

**Reply by fdelgadodelapuente** [reply]
Hi Eric,

Thanks for the detailed explanation. Do you know of a reference that explains why the full-aperture pixel spacing is small but actual azimuth resolution is much larger?

Francisco

**Reply by CunrenLiang** [reply]
That's because ALOS-2 ScanSAR is a multiple-look ScanSAR system. The number of looks is about four, which means each target on ground can be imaged by at least four bursts. In full-aperture product, all burst signals are added together. While single burst bandwidth is much smaller than PRF, the total bandwidth of all bursts combined is about PRF despite there are gaps (corresponding to burst gaps) in the whole band. So we still need a sampling rate as large as PRF, and the pixel size is thus small.

Actually with full-aperture product, we can also get azimuth pixel offset results comparable to stripmap mode. It may be a little noisier. The main issue is ionosphere. But we have alos2burstApp.py that can generate much better azimuth deformation using MAI, so there is not a big need for azimuth offset.

**Comment by EJFielding** [comment]
@fdelgadodelapuente There is a good explanation in Cunren's first paper on ALOS-2 ScanSAR:
Liang, C. & Fielding, E.J., 2016. Interferometric Processing of ScanSAR Data Using Stripmap Processor: New Insights From Coregistration, IEEE Transactions on Geoscience and Remote Sensing, 54, 4343-4354.
